nemo_skills/evaluation/metrics/speechlm_metrics.py

```diff
@@ -0,0 +1,181 @@
+
+        if category == "open" and "judge_correct" in score_dict:
+            score_dict["correct"] = score_dict["judge_correct"]
+        elif "is_correct" in prediction:
+            score_dict["correct"] = prediction["is_correct"]
+        else:
+            score_dict["correct"] = False
+
+        return score_dict
+
+    def get_incorrect_sample(self, prediction: dict) -> dict:
+        """Return a sample marked as incorrect for all metrics."""
+        prediction = prediction.copy()
+        prediction["is_correct"] = False
+        prediction["judge_correct"] = False
+        if not prediction.get("generation", "").strip():
+            prediction["generation"] = None
+        return prediction
+
+    def update_common_metrics(self, agg_dict):
+        """Override to always include avg_tokens even if 0 since it's in metrics_to_print."""
+        agg_dict["num_entries"] = self.total
+        agg_dict["avg_tokens"] = int(self.avg_tokens / self.total) if self.total > 0 else 0
+        if self.max_end_time > float("-inf") and self.min_start_time < float("inf"):
+            agg_dict["gen_seconds"] = int(self.max_end_time - self.min_start_time)
+
+    def update(self, predictions):
+        """Update metrics with new predictions."""
+        super().update(predictions)
+
+        predicted_answers = [pred.get("generation", None).strip() or None for pred in predictions]
+
+        self._compute_pass_at_k(predictions=predictions, predicted_answers=predicted_answers)
+        self._compute_majority_at_k(predictions=predictions, predicted_answers=predicted_answers)
+
+    def get_metrics(self):
+        """Get computed metrics."""
+        metrics_dict = super().get_metrics()
+
+        for agg_mode, agg_metrics in metrics_dict.items():
+            if "no_answer" in agg_metrics:
+                agg_metrics["no_answer"] = agg_metrics["no_answer"] / 2.0
+
+            # Set success_rate based on correct field
+            if "correct" in agg_metrics:
+                agg_metrics["success_rate"] = agg_metrics["correct"]
+            elif "judge_correct" in agg_metrics:
+                agg_metrics["success_rate"] = agg_metrics["judge_correct"]
+
+        return metrics_dict
+
+    def evaluations_to_print(self):
+        """Specify which evaluation modes to print."""
+        evals = [f"pass@{self.max_k}"]
+        if self.max_k > 1:
+            evals.extend([f"majority@{self.max_k}", f"pass@1[avg-of-{self.max_k}]"])
+        return evals
+
+    def metrics_to_print(self):
+        """Specify which metrics to print."""
+        base_metrics = {
+            "avg_tokens": as_int,
+            "gen_seconds": as_int,
+            "success_rate": as_percentage,
+        }
+
+        if self.compute_no_answer:
+            base_metrics["no_answer"] = as_percentage
+
+        base_metrics["num_entries"] = as_int  # Add at end for better display order
+
+        return base_metrics
+
+
+def compute_score(combined_metrics: dict) -> dict:
```

Comment:
better to move this to a new file with mmau_pro in the name (can also place it inside dataset folder I guess)